hack/tools/check_version.sh

```diff
@@ -0,0 +1,17 @@
+#! /bin/bash
+
+# Takes path , program, version as input
+# exit 0 if the program, when called from path, has the version in its output.
+VERSION=`$1/$2 version` 
+
+# We dont check v bc, we dont care about "v0.40.0" vs "0.40.0" and some
+# programs print version strings out differently.
+if echo $VERSION | `grep -q $3 | sed 's/v//g'` ; then
```

Comment:
The backticks look weird here...
And I don't understand what the `sed` is trying to do?  Does it want to remove the `v` from the `$VERSION` variable?  If so maybe it should be line 5?

Reply:
ytt version returns "0.40.0" but the YTT_VERSION = "v0.40.0"

Reply:
ya , they basically say "dont bother matching the v" 


- `grep -q` will cause a failure if theres no match
- `sed 's/v//g' will replace "v0.40" with "0.40" 
- The final result: It will make sure that 

YTT_VERSION=v0.40 OR YTT_VERSION=0.40

both pass validation :)

Reply:
The check never fails for me:
```
$ make check_tools
# Check YTT Version or fail.
# Could do something more clever reusing versions in hack/tools/Makefile someday.
/Users/kmarc/git/tanzu-framework/hack/tools/check_version.sh /Users/kmarc/git/tanzu-framework/hack/tools/bin ytt v0.40.0
correct version *** 0.3.0 ***```